device/marvell/arm64-marvell_db98cx8580_16cd-r0/plugins/sfputil.py

```diff
@@ -0,0 +1,255 @@
+                # Leave the old code for backward compatibility
+                #if len(line.split()) >= 4:
+                #    fp_port_index = (line.split()[3])
+                #    print(fp_port_index)     
+                else:
+                    fp_port_index = portname.split("Ethernet").pop()
+                    fp_port_index = int(fp_port_index.split("s").pop(0))+1
+            else:  # Parsing logic for older 'portmap.ini' file
+                (portname, bcm_port) = line.split("=")[1].split(",")[:2]
+
+                fp_port_index = portname.split("Ethernet").pop()
+                fp_port_index = int(fp_port_index.split("s").pop(0))+1
+
+            if ((len(self.sfp_ports) > 0) and (fp_port_index not in self.sfp_ports)):
+                continue
+
+            if first == 1:
+                # Initialize last_[physical|logical]_port
+                # to the first valid port
+                last_fp_port_index = fp_port_index
+                last_portname = portname
+                first = 0
+
+            logical.append(portname)
+
+            logical_to_bcm[portname] = "xe" + bcm_port
+            logical_to_physical[portname] = [fp_port_index]
+            if physical_to_logical.get(fp_port_index) is None:
+                physical_to_logical[fp_port_index] = [portname]
+            else:
+                physical_to_logical[fp_port_index].append(
+                    portname)
+
+            if (fp_port_index - last_fp_port_index) > 1:
+                # last port was a gang port
+                for p in range(last_fp_port_index+1, fp_port_index):
+                    logical_to_physical[last_portname].append(p)
+                    if physical_to_logical.get(p) is None:
+                        physical_to_logical[p] = [last_portname]
+                    else:
+                        physical_to_logical[p].append(last_portname)
+
+            last_fp_port_index = fp_port_index
+            last_portname = portname
+
+            port_pos_in_file += 1
+
+        self.logical = logical
+        self.logical_to_bcm = logical_to_bcm
+        self.logical_to_physical = logical_to_physical
+        self.physical_to_logical = physical_to_logical
+
+       
+        #print(self.logical_to_physical)
+	'''print("logical: " + self.logical)
+        print("logical to bcm: " + self.logical_to_bcm)
+        print("logical to physical: " + self.logical_to_physical)
+        print("physical to logical: " + self.physical_to_logical)'''
+        #print("exiting port_tab_mappings")
+
+    @property
+    def port_start(self):
+        return self._port_start
+
+    @property
+    def port_end(self):
+        return self._port_end
+	
+    @property
+    def qsfp_ports(self):
+        return self._qsfp_ports
+
+    @property 
+    def port_to_eeprom_mapping(self):
+         return self._port_to_eeprom_mapping
+    
+    @property
+    def get_transceiver_change_event(self):
+        raise NotImplementedError 
```

Comment:
Why is `get_transceiver_change_event()` not implemented for any of the platforms in this PR?

Reply:
Change Interrupt line was not provided in HW/CPLD.